### PR TITLE
Notebook for subsetting p-type data

### PR DESCRIPTION
**Pull request by djgagne**
I have added a notebook for subsetting p-type data for uploading to zenodo and making the dataset more user friendly.

### CODE REVIEW

**Comment by djgagne** [comment]
I just updated layers to keras 3 and made some changes to the readme.

File: mlguess/keras/layers.py [hunk 1]
```diff
@@ -1,64 +1,106 @@
...
-    def __init__(self, units, name, eps=1e-12, **kwargs):
-        super(DenseNormalGamma, self).__init__(name=name, **kwargs)
+    def __init__(self, units: int,
+                 spectral_normalization: bool = False,
+                 eps: float = 1e-12, **kwargs):
+        super().__init__(**kwargs)
         self.units = int(units)
-        self.dense = tfa.layers.SpectralNormalization(tf.keras.layers.Dense(4 * self.units, activation=None))
-        #self.dense = tf.keras.layers.Dense(4 * self.units, activation=None)
+        self.NUM_OUTPUT_PARAMS = 4
```
Comment (lucas-erlandson-nvidia) [inline mlguess/keras/layers.py:26]: If NUM_OUTPUT_PARAMS doesn't change instance to instance you may want to make it a static variable

File: mlguess/keras/losses.py [hunk 2]
```diff
@@ -4,7 +4,15 @@
 
 
 class DirichletEvidentialLoss(tf.keras.losses.Loss):
-    def __init__(self, callback=False, name="dirichlet", this_epoch_num = None, class_weights=None):
+    """
+    Loss function for an evidential categorical model.
+    Args:
+        callback (list): List of callbacks.
```
Comment (lucas-erlandson-nvidia) [inline mlguess/keras/losses.py:10]: PEP 484 provides type annotations syntax

File: mlguess/keras/models.py [hunk 3]
```diff
@@ -1299,3 +1300,243 @@ def locate_best_model(filepath, metric="val_ave_acc", direction="max"):
...
+        adam_beta_2=0.999,
+        epsilon=1e-7,
+        decay=0,
+        verbose=0,
+        random_state=1000,
+        callbacks=[],
+        balanced_classes=0,
+        steps_per_epoch=0,
+        **kwargs):
+        super().__init__(**kwargs)
```
Comment (lucas-erlandson-nvidia) [inline mlguess/keras/models.py:1362]: Perhaps reorder initialization to match parameter ordering.

File: mlguess/keras/models.py [hunk 4]
```diff
@@ -1299,3 +1300,243 @@ def locate_best_model(filepath, metric="val_ave_acc", direction="max"):
...
+        model_class = cls(**conf["model"])
+        model_class.build_neural_network(len(input_features), len(output_features))
+        model_class.model.load_weights(weights)
+
+        # Load the path to ensemble weights
+        model_class.ensemble_member_files = []
+        if conf["ensemble"]["n_splits"] > 1:
+            for j in range(conf["ensemble"]["n_splits"]):
+                model_class.ensemble_member_files.append(
+                    os.path.join(conf["save_loc"], "models", f"model_{j}.h5")
```
Comment (lucas-erlandson-nvidia) [inline mlguess/keras/models.py:1461]: You mentioned .keras vs .h5 distinction, does that apply here?

File: mlguess/keras/layers.py (hunk 1 above)
Comment (charlie-becker) [reply mlguess/keras/layers.py:26]: Thanks for adding that as a variable, much cleaner to read and understand.

File: mlguess/keras/models.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import os
 import sys
 import glob
+import keras
 import numpy as np
 import pandas as pd
```
Comment (charlie-becker) [inline mlguess/keras/models.py:6]: Need to remove this tf dependancy./

**Review by charlie-becker — CHANGES_REQUESTED** [review]
looks good. After tf dependency is removed / resolved from `models.py`, should be good to go.

**Comment by charlie-becker** [comment]
After discussion, we will be converting `models.py` specifically to not depend on tensorflow in a future PR.

**Review by charlie-becker — APPROVED** [review]

**Pull request merged**